diff --git a/dwm-6.2/dwm.c b/dwm-6.2_/dwm.c
index 3e361fa..2d44c62 100644
--- a/dwm.c
+++ b/dwm.c
@@ -222,6 +222,7 @@ static void resizemouse(const Arg *arg);
 static void resizerequest(XEvent *e);
 static void restack(Monitor *m);
 static void run(void);
+static void runAutostart(void);
 static void scan(void);
 static int sendevent(Window w, Atom proto, int m, long d0, long d1, long d2, long d3, long d4);
 static void sendmon(Client *c, Monitor *m);
@@ -1549,6 +1550,12 @@ run(void)
 			handler[ev.type](&ev); /* call handler */
 }
 
+void
+runAutostart(void) {
+	system("cd ~/.dwm; ./autostart_blocking.sh");
+	system("cd ~/.dwm; ./autostart.sh &");
+}
+
 void
 scan(void)
 {
@@ -2494,6 +2501,7 @@ main(int argc, char *argv[])
 		die("pledge");
 #endif /* __OpenBSD__ */
 	scan();
+	runAutostart();
 	run();
 	cleanup();
 	XCloseDisplay(dpy);
